pinot-core/src/main/java/org/apache/pinot/core/plan/SelectionPlanNode.java

```diff
@@ -24,52 +24,65 @@
 import org.apache.pinot.core.indexsegment.IndexSegment;
 import org.apache.pinot.core.operator.query.EmptySelectionOperator;
 import org.apache.pinot.core.operator.query.SelectionOnlyOperator;
+import org.apache.pinot.core.operator.query.SelectionOperator;
 import org.apache.pinot.core.operator.query.SelectionOrderByOperator;
 import org.apache.pinot.core.query.selection.SelectionOperatorUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 /**
- * The <code>SelectionPlanNode</code> class provides the execution plan for selection query on a single segment.
+ * The <code>SelectionPlanNode</code> class provides the execution plan for selection query on a
+ * single segment.
  */
 public class SelectionPlanNode implements PlanNode {
-  private static final Logger LOGGER = LoggerFactory.getLogger(SelectionPlanNode.class);
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(SelectionPlanNode.class);
+  public static boolean enableUDFInSelection = true;
   private final IndexSegment _indexSegment;
   private final Selection _selection;
-  private final ProjectionPlanNode _projectionPlanNode;
+  private TransformPlanNode _transformPlanNode;
+  private ProjectionPlanNode _projectionPlanNode;
 
   public SelectionPlanNode(IndexSegment indexSegment, BrokerRequest brokerRequest) {
     _indexSegment = indexSegment;
     _selection = brokerRequest.getSelections();
-
-    if (_selection.getSize() > 0) {
-      int maxDocPerNextCall = DocIdSetPlanNode.MAX_DOC_PER_CALL;
-
-      // No ordering required, select minimum number of documents
-      if (!_selection.isSetSelectionSortSequence()) {
-        maxDocPerNextCall = Math.min(_selection.getOffset() + _selection.getSize(), maxDocPerNextCall);
+    if (enableUDFInSelection) {
```

Comment:
This can be removed, since it is hard-coded to be true?

Reply:
Done. Removed it along with old classes